packages/react-components/react-persona/Spec.md

```diff
@@ -1,63 +1,279 @@
+```
+
+Persona with Avatar + PresenceBadge:
+
+```jsx
+<Persona
+  primaryText="Kevin Sturgis"
+  secondaryText="Software Engineer"
+  tertiaryText="Offline"
+  avatar={{ name: 'Kevin Sturgis' }}
+  badge={{ status: 'offline', outOfOffice: true }}
+/>
+```
+
+Persona with icon:
+
+```jsx
+<Persona primaryText="Person Call Icon" icon={<PersonCallRegular />} />
+```
+
+Persona with PresenceBadge:
+
+```jsx
+<Persona primaryText="Kevin Sturgis" badge={{ status: 'offline', outOfOffice: true }} />
+```
+
+Persona with image:
+
+```jsx
+<Persona
+  primaryText="Katri Athokas"
+  secondaryText="Software Engineer"
+  tertiaryText="CXE"
+  quaternaryText="Offline"
+  image={{ src: 'https://fabricweb.azureedge.net/fabric-website/assets/images/avatar/KatriAthokas.jpg' }}
+/>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+There are three alignment variants:
+
+- start: Coin on the left and text on the right.
+- center: Coin on top and text on the bottom.
+- end: Coin on the right and text on the left.
+
+There are 5 coin variats:
+
+- Badge
+- icon
+- Avatar
+- Avatar + Badge
+  - > Note: When there's an Avatar and a Badge and the Avatar provided already has a Badge, the provided Badge will take over the provided Badge.
+- image
+
+There are 2 sizing variants:
+
+- stretch: When the text lines have a larger height compared to the coin, stretch the coin to fit text lines height.
+- fixed: Keep the coin the same size, no matter the height of the text lines.
+
+**⚠️Responsive text variants will be left out of the initial implementation due to the accessibility concerns, they are added to this spec to start a discussion⚠️**
+
+There are 4 responsive text variants:
+
+- truncate: simple text truncation when the text doesn't fit in the container.
+  - > Note: after a talk offline, it was brought up that this might be an accessibility concern. @smhighley could you confirm these scenarios are concerns for the text?
+- hidden: same as truncate, but has a default `min-width` of 36px. If the width is less than `min-width`, then the text is hidden.
+- wrap: simple text wrap.
+- lineclamp truncate: wraps text until specified number of lines and after those lines it truncates the text.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+**Slots**
+
+- `root`: The root slot for Persona.
+- `avatar`: The Avatar, if provided.
+- `badge`: The PresenceBadge, if provided.
+- `icon`: The icon, if provided.
+- `image`: The image, if provided.
+- `primaryText`: Primary text, this is the only required slot. I decided to have at least this slot required since it doesn't make sense to have just the coin without the text. ⚠️ Open to feedback though.
+- `secondaryText`: Secondary text, if provided.
+- `tertiaryText`: Tertiary text, if provided.
+- `quaternaryText`: Quaternary text, if provided.
```

Comment:
`Persona` as a name is opinionated (compared even to `ItemLayout), are there reasons no not use semantic naming like `title`, `description`, etc.?

Reply:
We decided to go with Persona over EntityLayout due to Entity/Item being too generic. Persona layout gives a little of context on what to expect of the component and is more familiar since v8 has the same name. For the semantic naming, there's no big reason. We could change it to semantic naming if it's preferred.

Reply:
> We decided to go with Persona over EntityLayout due to Entity/Item being too generic.

But we use too generic `*text` slots then 😸 IMO as we went with more semantic name (`Persona`) it's better to use semantic names for slots. `description` was just an example, feel free to suggest/use own options.

Reply:
How about `primarySlot`, `secondarySlot`, ...? I feel like primary, secondary, and so on should be kept, but I do agree that `Text` won't work. Especially if we want to use breadcrumbs and such.